### PR TITLE
samba: 4.11.5 -> 4.12.0

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Update to latest samba version.

[Changelog](https://www.samba.org/samba/history/samba-4.12.0.html)

###### Things done

Backported patch removed.

Tested as file server.

Samba test in `nixos/tests/samba.nix` sucessfully executed.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/samba/4.x.nix [hunk 1]
```diff
@@ -34,17 +34,12 @@ stdenv.mkDerivation rec {
     ./patch-source3__libads__kerberos_keytab.c.patch
     ./4.x-no-persistent-install-dynconfig.patch
     ./4.x-fix-makeflags-parsing.patch
-    (fetchpatch {
```
Comment (jtojnar) [inline pkgs/servers/samba/4.x.nix]: `fetchpatch` can be dropped from args.

File: pkgs/servers/samba/4.x.nix [hunk 2]
```diff
@@ -34,17 +34,12 @@ stdenv.mkDerivation rec {
...
-      url = "https://gitlab.com/samba-team/samba/commit/5e517e57c9d4d35e1042a49d3592652b05f0c45b.patch";
-      sha256 = "1bbldf794svsdvcbp649imghmj0jck7545d3k9xs953qkkgwkbxi";
-    })
   ];
 
   nativeBuildInputs = optionals stdenv.isDarwin [ rpcgen fixDarwinDylibNames ];
 
   buildInputs = [
-    python pkgconfig perl libxslt docbook_xsl docbook_xml_dtd_42 /*
+    python pkgconfig perl perl.pkgs.ParseYapp libxslt docbook_xsl docbook_xml_dtd_42 /*
```
Comment (jtojnar) [inline pkgs/servers/samba/4.x.nix]: This should likely go to `nativeBuildInputs`. Along with `pkgconfig perl libxslt docbook_xsl docbook_xml_dtd_42 /* docbook_xml_dtd_45 */`.

Comment (bachp) [reply pkgs/servers/samba/4.x.nix]: I tought that too. But wasn't sure.

File: pkgs/servers/samba/4.x.nix (hunk 1 above)
Comment (bachp) [reply pkgs/servers/samba/4.x.nix]: done

**Comment by bachp** [comment]
@risicle @jtojnar Is there anything I can do to get this merged?

**Comment by jtojnar** [comment]
Thanks for the update.

**Pull request merged**